_episodes/02-image-basics.md

```diff
@@ -97,54 +60,214 @@ color, but that each pixel can have a different color from its neighbors.
+## More colors
+Up to now, we only had a 2 color matrix, but we can have more if we use other numbers or fractions. One common way is to use the numbers between 0 and 255 to allow for 256 different colors or 256 different levels of grey.  Let's try that out.
+ ~~~
+#make a copy of eight
+three_colors = skimage.io.imread(fname="data/eight.tif")
+
+#multiply the whole matrix by 128
+three_colors = three_colors * 128
+
+# set the middle row (index 2) to the value of 255., so you end up with the values 0.,128.,and 255
+three_colors[2,:] = 255.
+fig, ax = plt.subplots()
+plt.imshow(three_colors)
+print(three_colors)
+~~~
+{: .language-python}
+![Image of three colors](../fig/02-three-colors.png)
+We now have 3 colors, but are they the three colors you expected?  They all appear to be on a continum of dark purple on the low end and yellow on the high end. This is a consequence of the default color map (cmap) in this library. You can think of a color map as an association or mapping of numbers to a specific color. However, the goal here is not to have one number for every color under the sun, but rather to have a continuum of colors that demonstrate relative intensity. In our specific case here for example, 255 or the highest intensity is mapped to yellow, and 0 or the lowest intensity is mapped to a dark purple. The best color map for your data will vary and there are many options built in, but this default selection was not arbitrary.  A lot of science went into making this the default due to its robustness when it comes to how the human mind interprets relative color values, grey-scale printability, and color-blind friendliness ([https://matplotlib.org/stable/tutorials/colors/colormaps.html](https://matplotlib.org/stable/tutorials/colors/colormaps.html), [https://bids.github.io/colormap/](https://bids.github.io/colormap/)).  Thus it is a good place to start, and you should change it only with purpose and forethought.  For now, let's see how you can do that using an alternative map you have likley seen before where it will be even easier to see it as a mapped continuum of intensities: greyscale.
+ 
+ ~~~
+fig, ax = plt.subplots()
+plt.imshow(three_colors,cmap=plt.cm.gray)
+~~~
+{: .language-python}
+![Image in greyscale](../fig/02-grayscale.png)
+Above we have exactly the same underying data matrix, but in greyscale.  Zero maps to black, 255 maps to white, and 128 maps to medium grey.
+
+## Even More Colors
+
+This is all well and good at this scale, but what happens when we instead have a picture of a natural landscape that contains millions of colors.  Having a one to one mapping of number to color like this would be inefficient and make adjustments and building tools to do so very difficult.  Rather than larger numbers, the solution is to have more numbers in more dimensions. Let's see an example using a 4 X 4 matrix with 3 dimensions.  Rather than loading it from a file, we will generate an examples using numpy.
```

Comment:
Perhaps a few sentences explaining channels would be helpful here for learners to see the rationale or using more dimensions?

Reply:
Maybe a callout on color models?

Reply:
I think the start of the next chapter does a good job of exploring channels, so maybe it would be good to preview that as coming and maybe reference back to this in the start of the next chapter to connect those ideas

Reply:
Adjusted to this: This is all well and good at this scale, but what happens when we instead have a picture of a natural landscape that contains millions of colors. Having a one to one mapping of number to color like this would be inefficient and make adjustments and building tools to do so very difficult. Rather than larger numbers, the solution is to have more numbers in more dimensions. Storing the numbers in a multi-dimensional matrix where each color or property like transparency is associated with its own dimension allows for indiviudual contributions to a pixel to be adjusted independently. This ability to manipulate properties of groups of pixel separately will key to certain techniques explored in later chapters of this lesson. To get started let’s see an example of how different dimensions of information combine to produce a set of pixels using a 4 X 4 matrix with 3 dimensions for the colors red, green, and blue. Rather than loading it from a file, we will generate an examples using numpy.